src/osp/Active/SysVehicle.cpp

```diff
@@ -276,136 +192,144 @@ float SysVehicle::compute_hier_volume(ActiveScene& rScene, ActiveEnt part)
+        DependRes<Mesh> meshRes = glResources.get<Mesh>(meshData.name());
+
+        if (meshRes.empty())
+        {
+            // Mesh isn't compiled yet, compile it
+            meshRes = AssetImporter::compile_mesh(meshData, glResources);
+        }
 
-            //Mesh* mesh = nullptr;
-            Package& glResources = rScene.get_context_resources();
-            DependRes<Mesh> meshRes = glResources.get<Mesh>(
-                                            part.get_strings()[drawable.m_mesh]);
+        std::vector<DependRes<Texture2D>> textureResources;
+        textureResources.reserve(pcompDrawable.m_textures.size());
+        for (DependRes<ImageData2D> imageData : pcompDrawable.m_textures)
+        {
+            DependRes<Texture2D> texRes = glResources.get<Texture2D>(imageData.name());
 
-            if (meshRes.empty())
+            if (texRes.empty())
             {
-                // Mesh isn't compiled yet, now check if mesh data exists
-                std::string const& meshName = part.get_strings()[drawable.m_mesh];
-                DependRes<MeshData> meshData = package.get<MeshData>(meshName);
-                meshRes = AssetImporter::compile_mesh(meshData, glResources);
+                // Texture isn't compiled yet, compile it
+                texRes = AssetImporter::compile_tex(imageData, glResources);
             }
+            textureResources.push_back(texRes);
+        }
 
-            std::vector<DependRes<Texture2D>> textureResources;
-            textureResources.reserve(drawable.m_textures.size());
-            for (size_t i = 0; i < drawable.m_textures.size(); i++)
-            {
-                uint32_t texID = drawable.m_textures[i];
-                std::string const& texName = part.get_strings()[texID];
-                DependRes<Texture2D> texRes = glResources.get<Texture2D>(texName);
+        // by now, the mesh and texture should both exist
 
-                if (texRes.empty())
-                {
-                    DependRes<ImageData2D> imageData = package.get<ImageData2D>(texName);
-                    texRes = AssetImporter::compile_tex(imageData, glResources);
-                }
-                textureResources.push_back(texRes);
-            }
+        ActiveEnt currentEnt = newEntities[pcompDrawable.m_entity];
 
-            // by now, the mesh and texture should both exist
-            using adera::shader::Phong;
-            Phong::ACompPhongInstance shader;
-            shader.m_shaderProgram = glResources.get<Phong>("phong_shader");
-            shader.m_textures = std::move(textureResources);
-            shader.m_lightPosition = Vector3{10.0f, 15.0f, 5.0f};
-            shader.m_ambientColor = 0x111111_rgbf;
-            shader.m_specularColor = 0x330000_rgbf;
-            rScene.reg_emplace<Phong::ACompPhongInstance>(currentEnt, std::move(shader));
+        // TODO: Create components for generic properties of drawables:
+        //       ACompMesh, ACompSolid, ACompRoughnessTexture...
```

Comment:
This code still needs to be merged with the new SysRender code, which will take care of this